src/datasets/utils/py_utils.py

```diff
@@ -657,7 +657,9 @@ def _save_code(pickler, obj):
     #
     # Only those two lines are different from the original implementation:
     co_filename = (
-        "" if obj.co_filename.startswith("<") or obj.co_name == "<lambda>" else os.path.basename(obj.co_filename)
+        ""
+        if obj.co_filename.startswith("<") or "ipykernel" in obj.co_filename or obj.co_name == "<lambda>"
```

Comment:
Nice ! Can you also update the comments to explain how ipykernel functions are named ?

Also do you have a more robust way of detecting ipykernel functions ? If it comes from a filename equals to "path/to/ipykernel_folder/packaged/main.py" then it would trigger this "if" statement